networkdb/networkdb.go

```diff
@@ -380,24 +380,20 @@ func (nDB *NetworkDB) CreateEntry(tname, nid, key string, value []byte) error {
 // propagates this event to the cluster. It is an error to update a
 // non-existent entry.
 func (nDB *NetworkDB) UpdateEntry(tname, nid, key string, value []byte) error {
-	if _, err := nDB.GetEntry(tname, nid, key); err != nil {
+	entry, err := nDB.getEntry(tname, nid, key)
+	if err != nil {
 		return fmt.Errorf("cannot update entry as the entry in table %s with network id %s and key %s does not exist", tname, nid, key)
 	}
 
-	entry := &entry{
-		ltime: nDB.tableClock.Increment(),
-		node:  nDB.config.NodeID,
-		value: value,
-	}
+	// Update the entry
+	entry.ltime = nDB.tableClock.Increment()
+	entry.node = nDB.config.NodeID
```

Comment:
The `createOrUpdateEntry` call below used to be within a Lock/Unlock. Was the locking unnecessary? 

Is it possible that as the `entry` members (`ltime`/`node`/`value`) are being updated, a `getEntry` from a parallel thread might see inconsistent state for the `entry` ?

Reply:
the locking was to protect the internal data structure

Yes what you are saying is correct, and will affect both the update/delete code path changed here. 
At the same time, previous logic based on the acquisition of 2 times the same lock does not avoid the race between 2 creates for the same key.
Will try to come up with a unit test for that and provide a different fix